### ISSUE TITLE
Function template from base class is not considered if a more constrained candidate template is available in derived class

### ISSUE THREAD

**Issue by h-2** [llvm/llvm-project#58060]
```c++
#include <concepts>
#include <string>

struct S
{
    S() = default;
    S(auto) {}
};

struct T : S
{
    using S::S;

    T(std::integral auto) {}
};

int main()
{
    T t(std::string{});
}
```

This fails to build on clang++-15.0.0 and 16.0.0-++20220929042300+b55e76de8bcf-1~exp1~20220929042408.451. It builds on GCC.

Any of the following makes it build and select the constructor in the base class:
1. remove the constructor template in `T` completely
2. change the constructor template in `T` to not be a template (e.g. argument type `int`)
3. change the constructor template in `S` to not be a template and accept exactly `std::string`.

maybe related to #58059 ?

**Comment by h-2** [comment]
Since this is neither related to constructors nor `operator()`, I have changed the name.

Based on @erichkeane 's example in in #58059, the following should be an even simpler version:

```cpp
template<typename T>
concept Fails = false;

struct S
{
    void foo(auto) {}
};

struct S2 : S
{
    using S::foo;
    void foo(Fails auto) {}
};

int main()
{
    int i;
    S2 s2;
    s2.foo(i);
}
```
https://godbolt.org/z/5vT3dfdqK

**Comment by h-2** [comment]
Seems like this might be a duplicate of #50886?

At least the examples that both issues converge on are similar.

**Comment by usx95** [comment]
Fixed by 56099d242809f80984e4afa37693177484aab13d
